pkg/proxy/proxy.go

```diff
@@ -642,30 +638,13 @@ func (p *Proxy) removeRedirect(id string, wg *completion.WaitGroup) (err error,
 
 	implFinalizeFunc, implRevertFunc := r.implementation.Close(wg)
 
-	// Delay the release and reuse of the port number so it is guaranteed to be
-	// safe to listen on the port again. This can't be reverted, so do it in a
-	// FinalizeFunc.
-	proxyPort := r.listener.proxyPort
-	listenerName := r.listener.name
-
 	finalizeFunc = func() {
 		// break GC loop (implementation may point back to 'r')
 		r.implementation = nil
 
 		if implFinalizeFunc != nil {
 			implFinalizeFunc()
 		}
-
-		go func() {
-			time.Sleep(portReuseDelay)
-
-			proxyPortsMutex.Lock()
-			err := p.releaseProxyPort(listenerName)
```

Comment:
I take it that proxy port lifetime is now tied to the listener?

Reply:
This has been the case for a while now, redirects have shared proxy ports via shared listeners since about the time TPROXY was merged in.